easybuild/easyblocks/g/gcc.py

```diff
@@ -618,6 +619,52 @@ def post_install_step(self, *args, **kwargs):
                 raise EasyBuildError("Can't link '%s' to non-existing location %s", target, os.path.join(bindir, src))
 
+        # Rename include-fixed directory which includes system header files that were processed by fixincludes,
+        # since these may cause problems when upgrading to newer OS version.
+        # (see https://github.com/easybuilders/easybuild-easyconfigs/issues/10666)
+        glob_pattern = os.path.join(self.installdir, 'lib*', 'gcc', '*-linux-gnu', self.version, 'include-fixed')
+        matches = glob.glob(glob_pattern)
+        if matches:
+            if len(matches) == 1:
+                include_fixed_path = matches[0]
+
+                # limits.h and syslimits.h need to be copied to include/ first,
+                # these are strictly required (by /usr/include/limits.h for example)
+                include_path = os.path.join(os.path.dirname(include_fixed_path), 'include')
+                retained_header_files = ['limits.h', 'syslimits.h']
+                for fn in retained_header_files:
+                    from_path = os.path.join(include_fixed_path, fn)
+                    to_path = os.path.join(include_path, fn)
+                    if os.path.exists(from_path):
+                        if os.path.exists(to_path):
+                            raise EasyBuildError("%s already exists, not overwriting it with %s!", to_path, from_path)
+                        else:
+                            copy_file(from_path, to_path)
+                            self.log.info("%s copied to %s", from_path, to_path)
+                    else:
+                        self.log.warning("Can't copy non-existing file %s to %s", from_path, to_path)
+
+                readme = os.path.join(include_fixed_path, 'README.easybuild')
+                readme_txt = '\n'.join([
+                    "This directory was renamed by EasyBuild to avoid that the header files in it are picked up,",
+                    "since they may cause problems when the OS is upgraded to a new (minor) version.",
+                    '',
+                    "A couple of files were copied to %s first: %s" % (include_path, ', '.join(retained_header_files)),
+                    '',
+                    "See https://github.com/easybuilders/easybuild-easyconfigs/issues/10666 for more information.",
+                    '',
+                ])
+                write_file(readme, readme_txt)
+
+                include_fixed_renamed = include_fixed_path + '.renamed-by-easybuild'
+                move_file(include_fixed_path, include_fixed_renamed)
+                self.log.info("%s renamed to %s", include_fixed_path, include_fixed_renamed)
+            else:
+                raise EasyBuildError("Exactly one 'include-fixed' directory expected, found %d: %s",
+                                     len(matches), matches)
+        else:
+            self.log.info("No include-fixed subdirectory found at %s", glob_pattern)
+
     def sanity_check_step(self):
```

Comment:
I don't think we can realistically expect GCC to ever pass a sanity check any more...